api/dbadapters/presto.ts

```diff
@@ -0,0 +1,259 @@
+    const validationQueriesWithoutWrappers = collectEvaluationQueries(
+      queryOrAction,
+      useSingleQueryPerAction
+    );
+
+    const queryEvaluations = new Array<dataform.IQueryEvaluation>();
+    for (const { index, validationQuery } of validationQueries) {
+      let evaluationResponse: dataform.IQueryEvaluation = {
+        status: dataform.QueryEvaluation.QueryEvaluationStatus.SUCCESS
+      };
+      try {
+        await this.execute(validationQuery.query);
+      } catch (e) {
+        evaluationResponse = {
+          status: dataform.QueryEvaluation.QueryEvaluationStatus.FAILURE,
+          error: {
+            message: e.message,
+            errorLocation: {
+              line: e.errorLocation?.lineNumber,
+              column: e.errorLocation?.columnNumber
+            }
+          }
+        };
+      }
+      queryEvaluations.push(
+        dataform.QueryEvaluation.create({
+          ...evaluationResponse,
+          incremental: validationQuery.incremental,
+          query: validationQueriesWithoutWrappers[index].query
+        })
+      );
+    }
+    return queryEvaluations;
+  }
+
+  public async schemas(catalog: string): Promise<string[]> {
+    const result = await this.execute(`show schemas from ${catalog}`);
+    return flatten(result.rows);
+  }
+
+  public async createSchema(catalog: string, schema: string): Promise<void> {
+    await this.execute(`create schema if not exists ${catalog}.${schema}`);
+  }
+
+  public async tables(): Promise<dataform.ITarget[]> {
+    const catalogs = await this.catalogs();
+    let targets: dataform.ITarget[] = [];
+    await Promise.all(
+      catalogs.map(
+        async catalog =>
+          await Promise.all(
+            (await this.schemas(catalog)).map(async targetSchema => {
+              const result = await this.execute(`show tables from ${catalog}.${targetSchema}`);
+              targets = targets.concat(
+                flatten(result.rows).map(table =>
+                  dataform.Target.create({
+                    database: catalog,
+                    schema: targetSchema,
+                    name: table as string
+                  })
+                )
+              );
+            })
+          )
+      )
+    );
+    return targets;
+  }
+
+  public async table(target: dataform.ITarget): Promise<dataform.ITableMetadata> {
+    let columnsResult: IExecutionResult;
+    try {
+      // TODO: In the future, the connection won't specify the catalog (database) and schema to allow
+      // moving data between tables, in which case target resolution should throw if both are not present.
+      columnsResult = await this.execute(`describe ${resolveTarget(target)}`);
+    } catch (e) {
+      // If the table doesn't exist just return empty metadata.
+      // In this circumstance the error will be of type Presto.IPrestoClientError.
+      if (e.errorName === "TABLE_NOT_FOUND") {
+        return dataform.TableMetadata.create({});
```

Comment:
this should really return `null`, as per the other warehouses (otherwise various logic in the `build` and `run` steps will break in certain circumstances)